src/formatting.rs

```diff
@@ -559,7 +559,7 @@ impl<'a> FormatLines<'a> {
     fn char(&mut self, c: char, kind: FullCodeCharKind) {
         self.newline_count = 0;
         self.line_len += if c == '\t' {
-            self.config.tab_spaces()
+            self.config.tab_spaces() - self.line_len % self.config.tab_spaces()
         } else {
             1
```

Comment:
This `1` can also cause issues with non-ascii idents. Could be `c.len_utf8()`.

Reply:
Another possible fix for tabs is to use `self.line_buffer.push_str(&" ".repeat(self.config.tab_spaces()))`. Not sure which is correct.